src/main/java/com/coveros/selenified/services/Assert.java

```diff
@@ -0,0 +1,167 @@
+ * troubleshooting and debugging failing tests.
+ *
+ * @author Max Saperstone
+ * @version 3.2.0
+ * @lastupdate 4/4/2019
+ */
+public class Assert implements Check {
+
+
+    // this will be the name of the file we write all commands out to
+    private final Reporter reporter;
+
+    // this is the driver that will be used for all selenium actions
+    private final Response response;
+
+    /**
+     * The default constructor passing in the app and output file
+     *
+     * @param response - the response from the web services call
+     * @param reporter - the file to write all logging out to
+     */
+    public Assert(Response response, Reporter reporter) {
+        this.response = response;
+        this.reporter = reporter;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Reporter getReporter() {
+        return reporter;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Response getResponse() {
+        return response;
+    }
+
+    ///////////////////////////////////////////////////////
+    // assertions about the page in general
+    ///////////////////////////////////////////////////////
+
+    /**
+     * Asserts the actual response code is equals to the expected response
+     * code, and writes that out to the output file. If this fails, the code will
+     * immediately exit, and record the error.
+     *
+     * @param expectedCode - the expected response code
+     */
+    @Override
+    public void equals(int expectedCode) {
+        assertEquals("Code Mismatch", expectedCode, checkEquals(expectedCode));
+    }
+
+    /**
+     * Asserts the actual response json payload is equal to the expected
+     * response json payload, and writes that out to the output file. If this fails, the code will
+     * immediately exit, and record the error.
+     *
+     * @param expectedJson - the expected response json object
+     */
+    @Override
+    public void equals(JsonObject expectedJson) {
+        assertEquals("JsonObject Response Mismatch", expectedJson, checkEquals(expectedJson));
+    }
+
+    /**
+     * Asserts the actual response json payload is equal to the expected
+     * response json payload, and writes that out to the output file. If this fails, the code will
+     * immediately exit, and record the error.
+     *
+     * @param expectedJson - the expected response json array
+     */
+    @Override
+    public void equals(JsonArray expectedJson) {
+        assertEquals("JsonArray Response Mismatch", expectedJson, checkEquals(expectedJson));
+    }
+
+    /**
+     * Asserts the actual response payload is equal to the expected
+     * response payload, and writes that out to the output file. If this fails, the code will
+     * immediately exit, and record the error.
+     *
+     * @param expectedMessage - the expected response message
+     */
+    @Override
+    public void equals(String expectedMessage) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either override Object.equals(Object), or rename the method to prevent any confusion. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1201)